### ISSUE TITLE
Launchpad: Delete new launchpad/checklist endpoint

### ISSUE THREAD

**Issue by edanzer** [Automattic/wp-calypso#76368]
### Details

We [updated the main /launchpad endpoint to return the checklist](https://github.com/Automattic/jetpack/pull/30227), which means we no longer needed the newer launchpad/checklist  endpoint. However, that endpoint is being used in at least a few new temporary hooks while we wait backend code to be deployed via Jetpack. 

Once we've updated all instances using the new endpoint, we should follow up and delete that. 

At least the following two issues need to be done first, though there may be others. We'll just want to confirm the new endpoint is not being used anywhere before deleting it. 
   - https://github.com/Automattic/wp-calypso/issues/76362
   - https://github.com/Automattic/wp-calypso/issues/76363

**Comment by sixhours** [comment]
~~`useLaunchpadChecklist` was removed from the inline help component in https://github.com/Automattic/wp-calypso/pull/76596 so I'm closing this out!~~

**Comment by sixhours** [comment]
Oop, that comment was intended for [#76362](https://github.com/Automattic/wp-calypso/issues/76362), reopening :D
